### ISSUE TITLE
On macOS, VLC isn't automatically opened despite the app is installed in /Applications

### ISSUE THREAD

**Issue by baddles** [schwifty42069/piptv#5]
Like the title said, what would happen is that the script would go into a black screen with arrows waiting for the user to use Ctrl+C. However, the script can't open/play VLC, with no error whatsoever, even if VLC is opened or not.

**Comment by schwifty42069** [comment]
Unfortunately I can't test this with mac, but I think I have some volunteers who might, so give me a bit and I will get back to you!

**Comment by baddles** [comment]
No worries, I can try if you want me to.

**Comment by baddles** [comment]
Suggested from the pull request https://github.com/schwifty42069/piptv/pull/4 that --plugins_path syntax should be added. Here is the command on macOS:
if platform.system() == "Darwin":
            instance_args += "--plugins_path=/Applications/VLC.app/Contents/MacOS/plugins"
However, there is an issue after implementing this is that there is only sound, no VLC. I'm looking into this atm.

**Comment by schwifty42069** [comment]
Awesome great work, I will definitely do that. Keep me posted on that new issue that came up. I will leave this issue open until I add the additional instance args to vlc, and we iron the new issue out.

**Comment by z9t** [comment]
I am getting audio and no video on Mac Mojave?

**Comment by schwifty42069** [comment]
I am setting up a VM to do some testing for Mac so I will let you guys know if I figure anything out. Leaving this open for now

**Comment by Jumacbook** [comment]
Hi thanks for the work 😊 
Same here on Mojave : 
Sound : OK
Vidéo : KO

**Comment by schwifty42069** [comment]
Haha alright. It will be a bit before I can test it, because my internet here is garbage and the only iso I could find for MacOS is 6GB lol. As soon as I am able to test, I will let you guys know what I find!

**Comment by schwifty42069** [comment]
I haven't been able to test in macOS, but I did add a debug flag to the channel tune prompt. So if you want, type the channel name followed by --debug and paste the output here

for example, instead of typing just espn, type espn --debug and paste the output here

**Comment by wangw05** [comment]
Here's the debug log when trying to connect to tlc: [log](https://pastebin.com/mcAHLRE9)

**Comment by schwifty42069** [comment]
Damn, you're actually getting the stream data too, per the log. I should have a VM for macOS up by some point tonight, and will be able to test much more effectively. Will let you all know as soon as I do. Thanks for your patience!

**Comment by schwifty42069** [comment]
Can any of you play an HLS hotlink with VLC directly? So for example, try running:

vlc """http://peer3.savitar.tv/NFL/myStream/playlist.m3u8?wmsAuthSign=c2VydmVyX3RpbWU9OS8yMC8yMDE5IDExOjQyOjEyIFBNJmhhc2hfdmFsdWU9alNNSnFOS0xKNHdndUs0TDhHUmlhQT09JnZhbGlkbWludXRlcz0zNjAmaWQ9MA=="""

See if it plays successfully. I am just curious

**Comment by baddles** [comment]
![image](https://user-images.githubusercontent.com/33995791/65368539-ae475580-dc6c-11e9-870b-d947a9721f21.png)
Like this?

**Comment by CoachRocks** [comment]
If you mean directly in the app, the answer is no: 
<img width="1279" alt="Screen Shot 2019-09-21 at 01 19 44" src="https://user-images.githubusercontent.com/55593611/65369074-64c81d80-dc0e-11e9-99c1-88dbdf1cb8df.png">

Is that helpful?

**Comment by wangw05** [comment]
Instead of using: `vlc http://peer3.savitar.tv/NFL/myStream/playlist.m3u8?wmsAuthSign=c2VydmVyX3RpbWU9OS8yMC8yMDE5IDExOjQyOjEyIFBNJmhhc2hfdmFsdWU9alNNSnFOS0xKNHdndUs0TDhHUmlhQT09JnZhbGlkbWludXRlcz0zNjAmaWQ9MA==` to open a stream, Terminal requires the path to the VLC app itself.

Running `/Applications/VLC.app/Contents/MacOS/VLC http://peer3.savitar.tv/NFL/myStream/playlist.m3u8?wmsAuthSign=c2VydmVyX3RpbWU9OS8yMC8yMDE5IDExOjQyOjEyIFBNJmhhc2hfdmFsdWU9alNNSnFOS0xKNHdndUs0TDhHUmlhQT09JnZhbGlkbWludXRlcz0zNjAmaWQ9MA==` will bring up the VLC client and attempt to open the video stream. 

Resulting log 

> MacBook-Pro:~ username$ /Applications/VLC.app/Contents/MacOS/VLC http://peer3.savitar.tv/NFL/myStream/playlist.m3u8?wmsAuthSign=c2VydmVyX3RpbWU9OS8yMC8yMDE5IDExOjQyOjEyIFBNJmhhc2hfdmFsdWU9alNNSnFOS0xKNHdndUs0TDhHUmlhQT09JnZhbGlkbWludXRlcz0zNjAmaWQ9MA==
VLC media player 3.0.8 Vetinari (revision 3.0.8-0-gf350b6b5a7)
[00007f8c2d202560] main libvlc: Running vlc with the default interface. Use 'cvlc' to use vlc without interface.
[00007f8c2bddab60] access stream error: HTTP 403 error
2019-09-21 00:29:28.894 VLC[99392:30884330] Can't find app with identifier com.spotify.client
[00007f8c2bddab60] http stream error: error: HTTP/1.1 403 Forbidden

**Comment by schwifty42069** [comment]
Yeah the token expired, they only last 6 hours, I was hoping someone was gonna be active when I posted it haha. I am spinning up the macOS VM to do some testing. Sorry guys, I know this isn't fun.

**Comment by schwifty42069** [comment]
From the research I have done into this, it seems that vlcs python bindings may need to be built from source for them to function correctly on macOS. I am playing with it now, and if I get it to work building them from source I will document the method here so that others can use it. If anyone **really** wants to use it now, you can use it in an Ubuntu VM, or you can use the piptv-plex script on this repo to generate an m3u containing all of the channels with fresh tokens to be used in the media player of your choice.

**Comment by schwifty42069** [comment]
Have any of you tried the piptv-plex script or tried to run piptv in an ubuntu VM? Looks like it will be a bit before I can get an environment up and running for macOS, it's being a real pain in the ass -_-
Again, I'm sorry guys. I just whipped this thing up, didn't expect as many people to use it as there are!

**Comment by MBB417** [comment]
Just checking in. Have you been able to work on this issue at all? I'm trying to use piptv on macOS Catalina 10.15.1 and getting similar errors.
